qkernel/src/interrupt/aarch64/mod.rs

```diff
@@ -1,6 +1,137 @@
+use super::super::syscall_dispatch_aarch64;
```

Comment:
It's maybe better to move this to somewhere like /qlib/kernel/arch/aarch64/exception.rs

However I'm having some issues with module resolution. I have no idea why that happens:  when I put this file under qlib,  importing `syscall_dispatch_aarch64` from lib.rs causes compilation issue. Weirdly, `cargo check` has no error but `cargo build` does.

The only way I can make it work is to define this module directly in qkernel's mod tree, instead of under qlib. Any suggestion?

Reply:
What was path-chain in "lib.rs"?

To go a bit further, in my opinion naming/location is in general a deeper structural challenge/issue, e.g. in qlib/:

1. For easing porting the asm/aarch64.rs mirrors function(instriction)-names from asm/x86_64.rs.
2. qlib seems "overloaded", if an element is only used by either QKernel or QVisor then should be placed in either in qkernel/ or qvisor/.

Reply:
> What was path-chain in "lib.rs"?

I'm not sure what you mean by "path-chain" but here is a failed attempt to import from lib.rs

I tried to put the exception handler  in `/qkernel/src/qlib/kernel/arch/aarch64/exception.rs`

and do
```rust
use super::super::super::super::super::syscall_dispatch_aarch64;
// I can't recall if I used exactly 5x supers but I was certainly referring to the crate root.
```

I'll get error like "can't find syscall_..... in crate root".  But when I put the exception.rs in qkernel/interrupts/aarch64/mod.rs like I'm doing in this PR, everything works fine.

> qlib seems "overloaded" ......

I totally agree! I'm tempted to put the architectural bits under /qkernel . IMHO the qlib should only serve as an interface between the two parts.

Reply:
```
use super::super::super::super::super::syscall_dispatch_aarch64;
// I can't recall if I used exactly 5x supers but I was certainly referring to the crate root.```suggestion
```
As we already discussed this in the "real world", we can mark it as _resolved_.